### DISCUSSION TITLE
Odd highlighting/styling when using Prism as a NuxtJS plugin

### DISCUSSION THREAD

**Discussion by CodyBontecou** [PrismJS/prism#3102] [Q&A]
Here's how it's currently rendering:

<img width="406" alt="Screen Shot 2021-10-01 at 17 31 16" src="https://user-images.githubusercontent.com/20440899/135692770-65edd535-52bf-4a7f-9f78-c1d05bc39720.png">

It looks like it's being highlighted twice. Once in purple, once in grey, creating an odd 3d-like look.

Here's the code block I am using:

```js
import Prism from 'prismjs'
import Vue from 'vue'

// Include a theme:
import 'prismjs/themes/prism-tomorrow.css'

// Include language support: (optional)
import 'prismjs/components/prism-markdown'

// Include the toolbar plugin: (optional)
import 'prismjs/plugins/toolbar/prism-toolbar'
import 'prismjs/plugins/toolbar/prism-toolbar.css'

// Include the autolinker plugin: (optional)
import 'prismjs/plugins/autolinker/prism-autolinker'
import 'prismjs/plugins/autolinker/prism-autolinker.css'

// Include the line numbers plugin: (optional)
import 'prismjs/plugins/line-numbers/prism-line-numbers'
import 'prismjs/plugins/line-numbers/prism-line-numbers.css'

// Include some other plugins: (optional)
import 'prismjs/plugins/highlight-keywords/prism-highlight-keywords'
import 'prismjs/plugins/show-language/prism-show-language'
import 'prismjs/plugins/normalize-whitespace/prism-normalize-whitespace'

Vue.component('prism', {
  props: {
    lang: {
      type: String,
      default: 'js',
    },
  },
  mounted() {
    Prism.highlightAll()
  },
  template:
    '<div class="prism"><pre class="line-numbers" :class="`language-${lang}`" contenteditable><code><slot></slot></code></pre></div>',
})
```

**Comment by RunDevelopment** [comment]
The default theme uses a text-shadow like this. Where does the `text-shadow` property come from?

**Reply by CodyBontecou** [reply]
I'm not sure. My understanding was I'm using the theme `prism-tomorrow` with the line `import 'prismjs/themes/prism-tomorrow.css'
`

**Reply by RunDevelopment** [reply]
I meant when you inspect your page in the browser, where does the `text-shadow` come from?

**Reply by CodyBontecou** [reply]
Ah, sorry. It's from the class:

```css
code[class*="language-"], pre[class*="language-"]
```

**Reply by RunDevelopment** [reply]
Can you also see which file or is it mangled up by the build process?

Also, could you show me a screenshot of the whole rule `code[class*="language-"], pre[class*="language-"]`?

**Reply by CodyBontecou** [reply]
I think you're asking for this. It looks like I'm getting it from `prism.css` rather than `prismjs/themes/prism-tomorrow.css`

<img width="660" alt="Screen Shot 2021-10-01 at 19 29 39" src="https://user-images.githubusercontent.com/20440899/135697955-a309f3ad-2f4d-4d3b-ad5e-38cd50f8fa88.png">

**Reply by RunDevelopment** [reply]
So the problem is that you are using 2 themes at the same time. 

Why does this happen? I don't know enough about Vue and its build process to answer that. 

However, Vue's build process might be picking up on the [`styles` property in our `package.json`](https://github.com/PrismJS/prism/blob/a80a68ba507dae20f007a0817d9812f8eebcc5ce/package.json#L6). You might want to try temporarily removing that line in your local installation. If it really is the culprit, then you'll have to find a way to tell Vue to not use that CSS file.
